src/streaming/controllers/BufferController.js

```diff
@@ -347,17 +313,16 @@ MediaPlayer.dependencies.BufferController = function () {
 
         /* prune buffer on our own in background to avoid browsers pruning buffer silently */
         pruneBuffer = function() {
-            var bufferToPrune = 0,
+            var start = buffer.buffered.length ? buffer.buffered.start(0) : 0,
                 currentTime = this.playbackController.getTime(),
-                currentRange = this.sourceBufferExt.getBufferRange(buffer, currentTime);
-
-            // we want to get rid off buffer that is more than x seconds behind current time
-            if (currentRange !== null) {
-                bufferToPrune = currentTime - currentRange.start - MediaPlayer.dependencies.BufferController.BUFFER_TO_KEEP;
-                if (bufferToPrune > 0) {
-                    isPruningInProgress = true;
-                    this.sourceBufferExt.remove(buffer, 0, Math.round(currentRange.start + bufferToPrune), mediaSource);
-                }
+                bufferToPrune = currentTime - start - MediaPlayer.dependencies.BufferController.BUFFER_TO_KEEP;
+
+            // we want to get rid off buffer that is more than x
+            // seconds behind current time, but no pruning once it's
+            // finished.
+            if (!isPruningInProgress && mediaSource.readyState !== "ended" && bufferToPrune > MINIMUM_BUFFER_TO_PRUNE) {
```

Comment:
Just a question:
Are https://github.com/Eurosport/dash.js/commit/8e4bd1088467e1755ea3ebf31d78342236d6d595 
and 
https://github.com/MSOpenTech/dash.js/commit/d1a6739544abe1050b69f0ef47cc10b3661de4f1
scheduled for 1.5.2?